src/inference/sghmc.jl

```diff
@@ -32,10 +32,6 @@ mutable struct SGHMC{AD, T} <: StaticHamiltonian{AD}
     space::Set{T}    # sampling space, emtpy means all
 end
 SGHMC(args...) = SGHMC{ADBackend()}(args...)
-function SGHMC{AD}(learning_rate::Float64, momentum_decay::Float64, space...) where AD
```

Comment:
Why did we remove the limited `space` constructor? Isn't this used by Gibbs?

Reply:
See motivation here: https://turingjl.slack.com/archives/CBUHPCAH1/p1555772916045000